doc/specs/#4066 - Theme-controlled color scheme switch.md

```diff
@@ -0,0 +1,79 @@
+---
+author: Ítalo Masserano arkthur/italo.masserano@gmail.com
+created on: 2022-03-02
+last updated: 2022-03-02
+issue id: 4066
+---
+
+# Theme-controlled color scheme switch
+
+## Abstract
+
+The idea is for Windows Terminal to change automatically its color schemes according to what theme is selected, including the case where `system` theme is selected.
+
+## Inspiration
+
+I work remotely as a developer, so I have to spend a lot of hours in front of my PC screen. In my setup, right behind my desk I have a window, which is the only source of natural sunlight in my room.
+
+Normally I like dark modes in all the programs and apps I use, but when there's too much sunlight, it becomes annoying, and sometimes even painful, to work in dark mode. So, I have all the programs and apps I use (at least, those that can) set to switch their color themes to what the system has.
+
+The company I work for sent me a Macbook Pro, and my personal phone is an Android, both with automatic dark mode at sunset and light mode at sunrise, and in those devices it's been working relatively well. In Windows, as it is known, there's no such feature, so I manually change between dark and light mode when it's needed, and most of the programs and apps I use go along with this change. Windows Terminal, is not one of them.
+
+The theme changes just as expected, but in an app like this, this change only affects the top of the window, leaving almost all of the screen at the mercy of what the color scheme is, and it doesn't depend on the theme, which defeats any attempt to make a good use of the `system` theme feature.
+
+## Solution Design
+
+Could be implemented in the form of:
+
+```json
+"colorScheme": {
+    "light": "BlulocoLight",
+    "dark": "BlulocoDark"
+}
+```
```

Comment:
Team consensus was that both were good, but we were drawn to this implementation more. We could encapsulate the color scheme in the profile as something like a `ColorSchemeEntries` class, or heck even just a winrt `IMap<hstring, hstring>`, and look up the one we needed as necessary.

Reply:
I agree. I like this one better too ;)